pages/login.tsx

```diff
@@ -0,0 +1,50 @@
+import { useEffect, useState } from "react";
+import MainWrapper from "../components/MainContainer";
+import { useDispatch, useSelector } from "react-redux";
+import { AppState } from "../store/store";
+import { LogIn } from "../store/authSlice";
+import Router from "next/router";
+
+const Login = ()=>{
+    const [username, setUsername] = useState("")
+    const [password, setPassword] = useState("")
+    const [click, setClick] = useState(false);
+    const loggedIn = useSelector((state:AppState)=> state.auth.authState)
+    const dispatch = useDispatch()
+    function clickHandler(){
+        setClick(true)
+        dispatch(LogIn({username, password}))
+    }
+
+    useEffect(()=>{
+        if(loggedIn){
+            setClick(false)
+            Router.push('/')
+        }
+    }, [loggedIn])
+
+    return(
+        <MainWrapper>
+        <div style={{ flexDirection: "column", width:"100%" }}>
+        <div className={"base_padding content login"}>
+        <h1>Log in page</h1>
+        <div>
+        <label>Username: </label>
+            <input value={username} placeholder="username" onInput={(e)=>{setUsername(e.target.value);setClick(false)}} type="text" />
```

Comment:
Please extract all event handles to separate functions, don't write them inline
especially the ones which contain more than 1 statement
```suggestion
            <input value={username} placeholder="username" onInput={handleChange} type="text" />
```